app/routes.py

```diff
@@ -92,10 +92,18 @@ def create_shortlink():
         abort(make_error_msg(403, "Not Allowed"))
     response_headers = base_response_headers
     try:
-        url = request.json.get('url', None)
+        # Sometimes, the json parameter would be interpreted as a string, sometimes as a dict.
+        # We ensure we get what we want out of this parameter.
+        if isinstance(request.json, str):
```

Comment:
I which use case do we receive a valid json string ? Seems to me very strange that Flask would receive a valid JSON payload and return a string in the `.json` ! Might not be due to a wrong `content_type` in the request ?

Reply:
Yes, I'd also prefer to know and fix the reason behind this over using this work-around, if possible.

Reply:
I don't really know the why, 
When I run the tests as unit tests, I receive the json as a dictionary. At that point in the implementation, I simply had a line stating ```shorturl = request.json.get('shorturl')```
When I run the e2e tests, I receive the json as a string. this caused the following error : ```AttributeError: str object has no attribute 'get' ```
[request.json](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.json) is a shortcut to use get_json with default parameters.
[request.get_json()](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.get_json) default parameters state two comportment in our case : 1) we don't enforce json parsing (force = False) and 2) failure to parse the received json is not silenced. This means that we only try to parse json if the mime/type is specified as json, and according to the documentation, if the parsing is not silenced, the call to [on_json_loading_failed()](https://flask.palletsprojects.com/en/1.1.x/api/#flask.Request.on_json_loading_failed) should return a bad request as the default implementation

What I get out of this is that, for some reason, my json data could be read, understood, considered a json and given to me as a string, while still being a valid usable json (my proof being that e2e tests go through without a hitch)

Since I kinda don't want to change flask's comportment under the hood, I preferred to adopt a method that would takes both valid form of json I could receive and make sure I have what's necessary.

Reply:
If the content_type is wrong, the get_json() function returns ```None```

Reply:
And you can find the e2e test that created the situation here : https://github.com/geoadmin/infra-e2e-tests/blob/6e25aecffa732dc00b0475e4e406d2f22a3e664b/tests/test_service_shortlink/test_shortlink.py#L36-L40